extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/radio/FrontierSiliconRadioConnection.java

```diff
@@ -50,7 +50,7 @@
     private String sessionId;
 
     /** http clients, store cookies, so it is kept in connection class. */
-    private HttpClient httpClient = null;
+    private static final HttpClient HTTP_CLIENT = new HttpClient();
```

Comment:
No, it doesn't have to be. 

OK, in this case I will try to address the issue with the multiple instances with as little as possible changes in the handler.